app/controllers/teams_controller.rb

```diff
@@ -1,7 +1,8 @@
 class TeamsController < ApplicationController
   include ChangeDescription
 
-  before_action :set_team, only: [:show, :update]
+  before_action :set_team, only: [:show, :update, :typeahead]
+  before_action :set_query, only: [:typeahead]
```

Comment:
Since this is only used for the `typeahead` method, I think that you don't really need to do this in a filter.